### PR TITLE
Update plugin parent POM and BOM

### PR DESCRIPTION
**Pull request by basil**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1,20 +1,18 @@
...
     </parent>
     <properties>
         <revision>1.208</revision>
         <changelist>-SNAPSHOT</changelist>
-        <gitHubRepo>jenkinsci/sauce-ondemand-plugin</gitHubRepo>
+        <gitHubRepo>jenkinsci/${project.artifactId}-plugin</gitHubRepo>
         <!-- Baseline Jenkins version you use to build the plugin. Users must have this version or newer to run. -->
-        <jenkins.version>2.235.5</jenkins.version>
-        <!-- Java Level to use. Java 7 required when using core >= 1.612 -->
-        <java.level>8</java.level>
```
Comment (basil) [inline pom.xml:16]: Removed without replacement in a recent plugin parent POM.

File: pom.xml [hunk 2]
```diff
@@ -24,11 +22,10 @@
     <version>${revision}${changelist}</version>
     <packaging>hpi</packaging>
     <name>Jenkins Sauce OnDemand plugin</name>
-    <description>A Jenkins plugin to help create and teardown Sauce Labs tests and Sauce Connect sessions</description>
```
Comment (basil) [inline pom.xml:27]: Moved to `src/main/resources/index.jelly` in a recent version of Maven HPI plugin.

File: pom.xml [hunk 3]
```diff
@@ -24,11 +22,10 @@
...
-    <url>https://github.com/jenkinsci/sauce-ondemand-plugin</url>
+    <url>https://github.com/jenkinsci/${project.artifactId}-plugin</url>
 
     <scm>
-        <connection>scm:git:git://github.com/${gitHubRepo}.git</connection>
```
Comment (basil) [inline pom.xml:31]: Deprecated recently by GitHub.

File: pom.xml [hunk 4]
```diff
@@ -62,19 +59,19 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-antrun-plugin</artifactId>
-                <version>1.8</version>
+                <version>3.1.0</version>
```
Comment (basil) [inline pom.xml:62]: Latest upstream version.

File: pom.xml [hunk 5]
```diff
@@ -62,19 +59,19 @@
...
                 <executions>
                     <execution>
                         <phase>generate-sources</phase>
                         <configuration>
-                            <tasks>
+                            <target>
```
Comment (basil) [inline pom.xml:67]: Adapting to the latest upstream version.

File: pom.xml [hunk 6]
```diff
@@ -117,243 +114,160 @@
                 </exclusion>
             </exclusions>
         </dependency>
-
-        <!-- https://mvnrepository.com/artifact/com.google.guava/guava -->
-        <dependency>
-            <groupId>com.google.guava</groupId>
-            <artifactId>guava</artifactId>
-            <version>19.0</version>
-        </dependency>
```
Comment (basil) [inline pom.xml:126]: Removed the Guava dependency in `src/main/java` later in this PR to make this forward compatible with the Guava upgrade in Jenkins 2.332.x.

File: pom.xml [hunk 7]
```diff
@@ -117,243 +114,160 @@
...
-
-        <dependency>
-            <groupId>xml-apis</groupId>
-            <artifactId>xml-apis</artifactId>
-            <version>1.4.01</version>
-            <scope>test</scope>
-        </dependency>
```
Comment (basil) [inline pom.xml:133]: Unused.

File: pom.xml [hunk 8]
```diff
@@ -117,243 +114,160 @@
...
                 <exclusion>
                     <groupId>commons-codec</groupId>
                     <artifactId>commons-codec</artifactId>
                 </exclusion>
-            </exclusions>
-        </dependency>
-
-        <dependency>
-            <groupId>org.jenkins-ci</groupId>
-            <artifactId>crypto-util</artifactId>
```
Comment (basil) [inline pom.xml:236]: Unused.

File: pom.xml [hunk 9]
```diff
@@ -117,243 +114,160 @@
...
                 <exclusion>
                     <groupId>commons-io</groupId>
                     <artifactId>commons-io</artifactId>
                 </exclusion>
             </exclusions>
         </dependency>
-
         <dependency>
-            <groupId>org.jenkins-ci</groupId>
-            <artifactId>bytecode-compatibility-transformer</artifactId>
```
Comment (basil) [inline pom.xml:249]: Unused.

File: pom.xml [hunk 10]
```diff
@@ -117,243 +114,160 @@
...
-                    <artifactId>commons-io</artifactId>
+                    <groupId>javax.ejb</groupId>
+                    <artifactId>ejb</artifactId>
                 </exclusion>
-            </exclusions>
-        </dependency>
-
-        <dependency>
-            <groupId>commons-fileupload</groupId>
-            <artifactId>commons-fileupload</artifactId>
```
Comment (basil) [inline pom.xml:262]: Unused.

File: pom.xml [hunk 11]
```diff
@@ -117,243 +114,160 @@
...
-            <groupId>org.json</groupId>
-            <artifactId>json</artifactId>
-            <version>20220320</version>
+            <groupId>org.jenkins-ci.plugins</groupId>
+            <artifactId>credentials</artifactId>
         </dependency>
-
         <dependency>
-            <groupId>org.slf4j</groupId>
-            <artifactId>slf4j-jdk14</artifactId>
```
Comment (basil) [inline pom.xml:294]: Unused.

File: pom.xml [hunk 12]
```diff
@@ -117,243 +114,160 @@
...
-            <version>1.7.36</version>
+            <groupId>org.jenkins-ci.plugins</groupId>
+            <artifactId>jackson2-api</artifactId>
         </dependency>
-
         <dependency>
-            <groupId>org.slf4j</groupId>
-            <artifactId>slf4j-api</artifactId>
```
Comment (basil) [inline pom.xml:300]: Unused.

File: pom.xml [hunk 13]
```diff
@@ -117,243 +114,160 @@
...
+        <dependency>
+            <groupId>org.jenkins-ci.plugins.workflow</groupId>
+            <artifactId>workflow-step-api</artifactId>
         </dependency>
 
         <dependency>
-            <groupId>junit</groupId>
-            <artifactId>junit</artifactId>
-            <version>4.13.2</version>
-            <scope>test</scope>
```
Comment (basil) [inline pom.xml:328]: Provided in the plugin parent POM as of recent versions.

File: pom.xml [hunk 14]
```diff
@@ -117,243 +114,160 @@
...
-        <repository>
-            <id>saucelabs-repository</id>
-            <url>https://repository-saucelabs.forge.cloudbees.com/release</url>
-            <releases>
-                <enabled>true</enabled>
-            </releases>
-            <snapshots>
-                <enabled>true</enabled>
-            </snapshots>
-        </repository>
```
Comment (basil) [inline pom.xml:351]: Didn't seem to exist anymore and was confusing Enforcer.

File: src/main/java/com/saucelabs/jenkins/pipeline/ExpanderImpl.java [hunk 15]
```diff
@@ -3,7 +3,7 @@
 import hudson.EnvVars;
 import org.jenkinsci.plugins.workflow.steps.EnvironmentExpander;
 
-import javax.annotation.Nonnull;
+import edu.umd.cs.findbugs.annotations.NonNull;
```
Comment (basil) [inline src/main/java/com/saucelabs/jenkins/pipeline/ExpanderImpl.java:6]: Matching our preference to migrate away from JSR 305 throughout the plugin ecosystem.

File: src/main/java/com/saucelabs/jenkins/pipeline/SauceConnectStep.java [hunk 16]
```diff
@@ -127,7 +127,9 @@ public static final class DescriptorImpl extends StepDescriptor {
 
         @Override
         public Set<? extends Class<?>> getRequiredContext() {
-            return ImmutableSet.of(Run.class, Computer.class, TaskListener.class, SauceCredentials.class);
+            Set<Class<?>> context = new HashSet<>();
+            Collections.addAll(context, Run.class, Computer.class, TaskListener.class, SauceCredentials.class);
+            return Collections.unmodifiableSet(context);
```
Comment (basil) [inline src/main/java/com/saucelabs/jenkins/pipeline/SauceConnectStep.java:132]: Removing any usages of Guava for forward compatibility with 2.332.x.

File: src/test/java/hudson/plugins/sauce_ondemand/SauceOnDemandBuildActionTest.java [hunk 17]
```diff
@@ -34,10 +34,10 @@
...
 import org.json.JSONObject;
 import static org.junit.Assert.assertThat;
-import static org.mockito.Matchers.any;
-import static org.mockito.Matchers.anyInt;
-import static org.mockito.Matchers.anyList;
-import static org.mockito.Matchers.anyString;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.ArgumentMatchers.anyList;
+import static org.mockito.ArgumentMatchers.anyString;
```
Comment (basil) [inline src/test/java/hudson/plugins/sauce_ondemand/SauceOnDemandBuildActionTest.java:40]: Replacing deprecated usages with non-deprecated equivalents.

**Comment by alexh-sauce** [comment]
@basil I'm sorry I didn't see the notification for this PR earlier. Do you mind rebasing on the latest commit? I'd like to get your changes merged.

**Pull request merged**